R/Reporter.R

```diff
@@ -132,15 +141,15 @@ Reporter <- R6::R6Class( # nolint: object_name_linter.
     #'
     #' reporter$replace_card("Card1", card2)
     #' reporter$get_cards()[[1]]$get_name()
-    replace_card = function(id, card) {
-      if (is.character(id)) {
-        id <- which(names(private$cards) == id)
-      }
-      private$cards[[id]] <- card()
-      private$reactive_add_card(length(private$cards))
+    replace_card = function(card) {
+      checkmate::assert(private$check_append(card))
+      private$cards[[metadata(card, "id")]] <- card
+      private$hashes[[metadata(card, "id")]] <- rlang::hash(list(card, Sys.time()))
```

Comment:
Is hash here to detect if the card has been modified? If so, then maybe it is not needed:
- `ReportDocument` is an S3, when it is modified then its pointer changes
- `ReportCard` is R6 class and since it will be superseded by `ReportDocument` we don't need to support editor feature for it. 

Given above, in previewer-module we can detect if:
- card has been added or removed by detecting that names of the card list changed
- card has been modified when any card in Reporter is not identical to the `current_cards`